R/import_list.R

```diff
@@ -48,6 +48,7 @@ function(file,
         setclass <- NULL
     }
     if (length(file) > 1) {
+        names(file) <- gsub(paste0("\\.", tools::file_ext(file[1]), "$"), "", file, ignore.case = TRUE)
```

Comment:
I'm not sure I understand what's happening here. Can you explain?

Reply:
It's what the first comment in this PR is trying to explain but unfortunately the comment didn't attach itself to the precise line of code for clarity...

Also relates to Case 4 in the PR description

Reply:
I've since realised that assuming all files have the same extension somewhat defeats the purpose of ingesting multiple disparate datasets of differing file types at the same time.

```r
file <- c("mtcars.csv", "iris.csvy", "flights.xlsx", "USArrests.csv")
my_list <- import_list(file, rbind = FALSE)
names(my_list)
#> [1] "mtcars"  "iris.csvy"  "flights.xlsx"  "USArrests"
```

If it's going to strip file extensions it probably makes more sense to `lapply(file, tools::file_ext)` and strip each respective one so that we end up with

```r
exts <- paste0("\\.", lapply(file, tools::file_ext), "$")
names(file) <- sapply(seq_along(file), function(x) gsub(exts[x], "", file[x]))
names(file)
#> [1] "mtcars"  "iris"  "flights"  "USArrests"
```

Also, on second thoughts it also doesn't really seem appropriate to treat `zip` files differently so Case 3 should now be the same as Case 4

3. `zip` retains the file names *without* extension